proxy.php

```diff
@@ -202,34 +204,42 @@ function getHttpContentAndStatus($url, $timeout, $user_agent)
 
     $useFopen = @ini_get('allow_url_fopen') == '1';
 
-    $header = sprintf("Accept-Language: %s\r\n", str_replace(array("\n", "\t", "\r"), "", arrayValue($_SERVER, 'HTTP_ACCEPT_LANGUAGE', '')));
+    $header = [];
+    $header[] = sprintf("Accept-Language: %s", str_replace(array("\n", "\t", "\r"), "", arrayValue($_SERVER, 'HTTP_ACCEPT_LANGUAGE', '')));
 
     // NOTE: any changes made to Piwik\Plugins\PrivacyManager\DoNotTrackHeaderChecker must be made here as well
     if((isset($_SERVER['HTTP_X_DO_NOT_TRACK']) && $_SERVER['HTTP_X_DO_NOT_TRACK'] === '1')) {
-        $header .= "X-Do-Not-Track: 1\r\n";
+        $header[] = "X-Do-Not-Track: 1";
     }
 
     if((isset($_SERVER['HTTP_DNT']) && substr($_SERVER['HTTP_DNT'], 0, 1) === '1')) {
-        $header .= "DNT: 1\r\n";
+        $header[] = "DNT: 1";
+    }
+
+    if (isset($_SERVER['HTTP_COOKIE'])) {
+        $header[] = "Cookie: " . $_SERVER['HTTP_COOKIE'];
     }
 
     $stream_options = array('http' => array(
         'user_agent' => $user_agent,
         'header'     => $header,
-        'timeout'    => $timeout
+        'timeout'    => $timeout,
+
+        // 'ignore_errors' => true, // uncomment to debug error code responses
```

Comment:
Is it possible that this is the wrong way around? If commenting is for debugging, it shouldn't be uncommitted by default.

Reply:
It’s commented so developers don’t have to look it up next time when there is an error and zero output.